### DISCUSSION TITLE
Enhancement: Make Hitby/Nothitby take specific Hitdef Attributes

### DISCUSSION THREAD

**Discussion by ghost** [ikemen-engine/Ikemen-GO#896] [Feature Request]
For example, if you want to not be hit by an Air Normal Attack only, it can't happen. With how elecbyte implemented it, the sctrl will only affect either all air moves regardless of attack type, or affect all Normal attacks regardless of state type. It cannot specifically do both parameters. Combining hitby with nothitby also does not fix these issues. Possibly adding another parameter to the sctrl like "Attr:" so that it does not break compatibility with current implementation.

Ill paste what PlasmoidThunder said in the conversation we had in discord regarding this:

-(Not)HitBy is just bugged in general. No combination of the two lets you be specific in what inv is given due to how they work.

-Because by specifying S,NA, you'd expect it to give you invulnerability to all Standing Normal Attacks, not all Standing and all Normal Attacks.
You legit cannot tell MUGEN to give your character invulnerability to just air Normals.

-HitBy is bugged in the same way.
You can give yourself air inv with a NotHitBy that just has A as its value.
But you cannot be invulnerable to only Air Normal Attacks.
In the same sense that you can't give your character invulnerability to just Standing Throws.

**Comment by Windblade-GR01** [comment]
Thanks to the proposal.
We are planning to adding a new parameter to HitBy/NotHitBy that allows what you proposed.

**Reply by LasombraDemon** [reply]
> Thanks to the proposal. We are planning to adding a new parameter to HitBy/NotHitBy that allows what you proposed.

I also wanted to make players only hittable by certain *players*, and found no simple way to do this. A PlayerID parameter for HitBy/NotHitBy would also be awesome.

Then again, maybe just make a new SCTRL (to avoid compatibility issues) and give it a new name.

**Comment by ghost** [comment]
I actually was about to mention that Lasombra, good catch! I do think thats
a great idea as well, not hit by w specific player ID there are instances i
think this would be a nice addition
